cmake/PAL.cmake

```diff
@@ -394,21 +400,27 @@ function(ly_get_list_relative_pal_filename out_name in_name)
 
     # parent relative path is optional
     if(${ARGC} GREATER 4)
-        set(parent_relative_path ${ARGV4})
+        if(ARGV4)
+            set(parent_relative_path ${ARGV4})
+        endif()
     endif()
 
     # The Default object path for path is the LY_ROOT_FOLDER
     cmake_path(SET object_path NORMALIZE "${LY_ROOT_FOLDER}")
     if(${ARGC} GREATER 3)
-        # The user has supplied an object restricted path, the object path for consideration
-        cmake_path(SET object_path NORMALIZE ${ARGV3})
+        if(ARGV3)
+            # The user has supplied an object restricted path, the object path for consideration
+            cmake_path(SET object_path NORMALIZE ${ARGV3})
+        endif()
     endif()
 
     # The default restricted object path is O3DE_ENGINE_RESTRICTED_PATH
     cmake_path(SET object_restricted_path NORMALIZE "${O3DE_ENGINE_RESTRICTED_PATH}")
     if(${ARGC} GREATER 2)
-        # The user has supplied an object restricted path
-        cmake_path(SET object_restricted_path NORMALIZE ${ARGV2})
+        if(ARGV2)
+            # The user has supplied an object restricted path
+            cmake_path(SET object_restricted_path NORMALIZE ${ARGV2})
+        endif()
```

Comment:
I think we also need to surround the `o3de_pal_dir` call with double quotes around the `${object_restricted_path}`, `${object_path}` and `${parent_relative_path}` arguments to resolve them as an argument in case they are unset as variables during the call

Reply:
done